sunpy/coordinates/frames.py

```diff
@@ -587,6 +589,13 @@ def make_3d(self):
 
             d = np.fmin(d, dd) if self._spherical_screen['only_off_disk'] else dd
 
+        stack_trace = traceback.format_stack()
+        matching_string = 'wcsaxes.*_draw_grid'
+        if not any([re.search(matching_string, string) for string in stack_trace]) and np.all(np.isnan(d)) and np.any(np.isfinite(cos_alpha)):
```

Comment:
Why don't you make this line a lot shorter by doing the regex on the line above?

Reply:
Fixed.